File: Valid Mountain Array.go
func validMountainArray(arr []int) bool {
    n := len(arr)
    if n < 3 {
        return false
    }

    i := 0
    j := n-1
    for i < j && arr[i] < arr[i+1] {
        i++
    }

    for i < j && arr[j-1] > arr[j] {
        j--
    }

    return i == j && j != n-1 && i != 0
}

File: Valid Mountain Array.cpp
class Solution {
public:
    bool validMountainArray(vector<int>& arr) {
        int n = arr.size();
        if (n < 3) return false;

        int i = 0, j = n-1;
        while (i < j && arr[i] < arr[i+1]) i++;
        while (i < j && arr[j-1] > arr[j]) j--;

        return i == j && j != n-1 && i != 0;
    }
};

